src/App.js

```diff
@@ -14,11 +15,42 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    isShowed: false,
+  }
+
+  showList = () => {
+    this.setState(() => ({
+      isShowed: true,
+    }));
```

Comment:
тут не потрібний функціональний setState так як ти не використовуєш попередній state 
```suggestion
    this.setState({
      isShowed: true,
    });
```